### ISSUE TITLE
[MNG-5388] Restore embedded integration tests

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#6536]
**[Jason van Zyl](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jvanzyl)** opened **[MNG-5388](https://issues.apache.org/jira/browse/MNG-5388?redirect=false)** and commented

The embedded integration tests through not being run for a long time, and changes made in the verifier make it impossible for them to pass currently. They test long-lived processes using Maven and makes sure things are cleaned up correctly and they run an order of magnitude faster than the forked tests.

---
No further details from [MNG-5388](https://issues.apache.org/jira/browse/MNG-5388?redirect=false)

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

since the Apache Jenkins job configuration was having a timeout caused by JDK 5 (no idea why):
see [build 794 log](https://builds.apache.org/view/M-R/view/Maven/job/core-integration-testing-maven-3-embedded/794/console)

```
     [java] [main] INFO org.apache.maven.cli.MavenCli - Error stacktraces are turned on.
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Scanning for projects...
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - ------------------------------------------------------------------------
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Reactor Build Order:
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - 
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Apache Maven
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Maven Model
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Maven Artifact
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Maven Plugin API
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Maven Model Builder
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Maven Settings
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Maven Settings Builder
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Maven Repository Metadata Model
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Maven Aether Provider
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Maven Core
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Maven Compat
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Maven Embedder
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Maven Distribution
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger -                                                                         
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - ------------------------------------------------------------------------
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Building Apache Maven 3.1-SNAPSHOT
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - ------------------------------------------------------------------------
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - 
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - --- maven-clean-plugin:2.5:clean (default-clean) @ maven ---
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - 
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - --- maven-remote-resources-plugin:1.3:process (default) @ maven ---

BUILD FAILED
/home/jenkins/jenkins-slave/workspace/core-integration-testing-maven-3-embedded/maven-3-trunk/build.xml:253: Timeout: killed the sub-process

Total time: 10 minutes 48 seconds
Build step 'Invoke Ant' marked build as failure
Recording test results
Finished: FAILURE
```

I changed the JDK to version 6, which doesn't have this problem

then [there was a new problem in build 795 caused by Jenkins misconfiguration that used Maven 2.2.1 which cannot run the ITs](https://builds.apache.org/view/M-R/view/Maven/job/core-integration-testing-maven-3-embedded/795/console)]:

```
[java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - ------------------------------------------------------------------------
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Reactor Summary:
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - 
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Apache Maven ...................................... SUCCESS [14.600s]
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Maven Model ....................................... SUCCESS [4.837s]
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Maven Artifact .................................... SUCCESS [1.145s]
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Maven Plugin API .................................. SUCCESS [1.514s]
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Maven Model Builder ............................... SUCCESS [1.863s]
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Maven Settings .................................... SUCCESS [0.601s]
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Maven Settings Builder ............................ SUCCESS [0.725s]
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Maven Repository Metadata Model ................... SUCCESS [0.439s]
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Maven Aether Provider ............................. SUCCESS [1.580s]
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Maven Core ........................................ SUCCESS [12.450s]
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Maven Compat ...................................... SUCCESS [7.332s]
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Maven Embedder .................................... SUCCESS [1.791s]
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Maven Distribution ................................ SUCCESS [5.220s]
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - ------------------------------------------------------------------------
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - BUILD SUCCESS
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - ------------------------------------------------------------------------
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Total time: 54.511s
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Finished at: Sun Jan 06 16:41:41 UTC 2013
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - Final Memory: 25M/232M
     [java] [main] INFO org.apache.maven.cli.event.ExecutionEventLogger - ------------------------------------------------------------------------

maven-assembly:
     [echo] 
     [echo] The new Maven distribution was created as part of the MAVEN-COMPILE step, above.
     [echo] This goal just validates the presence of that distribution.

extract-assembly:
     [echo] Extracting assembly to /home/jenkins/jenkins-slave/workspace/core-integration-testing-maven-3-embedded/apache-maven-3-SNAPSHOT ...
   [delete] Deleting directory /home/jenkins/jenkins-slave/workspace/core-integration-testing-maven-3-embedded/apache-maven-3-SNAPSHOT
    [mkdir] Created dir: /home/jenkins/jenkins-slave/workspace/core-integration-testing-maven-3-embedded/apache-maven-3-SNAPSHOT
    [unzip] Expanding: /x1/jenkins/jenkins-slave/workspace/core-integration-testing-maven-3-embedded/maven-3-trunk/apache-maven/target/apache-maven-3.1-SNAPSHOT-bin.zip into /home/jenkins/jenkins-slave/workspace/core-integration-testing-maven-3-embedded/apache-maven-3-SNAPSHOT

all:

BUILD SUCCESSFUL
Total time: 1 minute 45 seconds
[core-integration-testing-maven-3-embedded] $ /bin/bash -xe /tmp/hudson8223322153174778776.sh
+ rm -rf /home/jenkins/jenkins-slave/workspace/core-integration-testing-maven-3-embedded/it-local-repo
[core-integration-testing-maven-3-embedded] $ mvn -f /home/jenkins/jenkins-slave/workspace/core-integration-testing-maven-3-embedded/core-integration-testing-trunk/pom.xml -Dmaven.home=/home/jenkins/jenkins-slave/workspace/core-integration-testing-maven-3-embedded/apache-maven-3-SNAPSHOT -Dmaven.repo.local=/home/jenkins/jenkins-slave/workspace/core-integration-testing-maven-3-embedded/it-local-repo clean install -Prun-its,embedded -B -U -V
Apache Maven 2.2.1 (r801777; 2009-08-06 19:16:01+0000)
Java version: 1.6.0_32
Java home: /x1/jenkins/tools/java/jdk1.6.0_32-32/jre
Default locale: en_US, platform encoding: ANSI_X3.4-1968
OS name: "linux" version: "3.2.0-35-generic" arch: "i386" Family: "unix"
[INFO] Scanning for projects...
Downloading: http://repo1.maven.org/maven2/org/apache/maven/maven-parent/22/maven-parent-22.pom
29K downloaded  (maven-parent-22.pom)
Downloading: http://repo1.maven.org/maven2/org/apache/apache/11/apache-11.pom
14K downloaded  (apache-11.pom)
[INFO] ------------------------------------------------------------------------
[ERROR] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] The projects in the reactor contain a cyclic reference: Edge between 'Vertex{label='org.apache.maven.its.plugins:maven-it-plugins'}' and 'Vertex{label='org.apache.maven.plugins:maven-plugin-plugin'}' introduces to cycle in the graph org.apache.maven.plugins:maven-plugin-plugin --> org.apache.maven.its.plugins:maven-it-plugin-core-stubs --> org.apache.maven.its.plugins:maven-it-plugins --> org.apache.maven.plugins:maven-plugin-plugin
```

I changed the Maven version from "(default)" (which selected 2.2.1) to "3.0.4"

the ITs now run, even if then still [cause failures in build 796](https://builds.apache.org/view/M-R/view/Maven/job/core-integration-testing-maven-3-embedded/796/console)

```
Résultats des tests (13 échecs / +13)
org.apache.maven.it.MavenITmng4387QuietLoggingTest.testit
org.apache.maven.it.MavenITmng4386DebugLoggingTest.testit
org.apache.maven.it.MavenITmng3951AbsolutePathsTest.testitMNG3951
org.apache.maven.it.MavenITmng3944BasedirInterpolationTest.testitMNG3944
org.apache.maven.it.MavenITmng3904NestedBuildDirInterpolationTest.testitMNG3904
org.apache.maven.it.MavenITmng3877BasedirAlignedModelTest.testitMNG3877
org.apache.maven.it.MavenITmng3864PerExecPluginConfigTest.testitMNG3864
org.apache.maven.it.MavenITmng3827PluginConfigTest.testitMNG3827
org.apache.maven.it.MavenITmng3760BaseUriPropertyTest.testitMNG3760
org.apache.maven.it.MavenITmng3760BaseUriPropertyTest.testitMNG3760SystemPropertyOverride
```

**Comment by jira-importer** [comment]
**[Hervé Boutemy](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hboutemy)** commented

next run only has 2 failures: https://builds.apache.org/view/M-R/view/Maven/job/core-integration-testing-maven-3-embedded/797/

```
	Résultats des tests (2 échecs / -11)
org.apache.maven.it.MavenITmng4387QuietLoggingTest.testit
org.apache.maven.it.MavenITmng4386DebugLoggingTest.testit
```

I don't know why this failure count has changed since the code nor configuration didn't change, but here it is

with these 2 isues fixed in [72bdc8602e5112aa273adc46b06d41c41f0f64a8](http://git-wip-us.apache.org/repos/asf/maven/diff/72bdc860), perhaps the next run will be clear of failures
